### PR TITLE
ceph: Compatibility for generating crds without sed

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
On mac the make crds command was failing on the sed commands. Now we remove the dependency on sed and just append the files to generate the helm resources

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: build/crds/build-crds.sh [hunk 1]
```diff
@@ -97,7 +97,6 @@ generating_main_crd
 
 for crd in "$OLM_CATALOG_DIR/"*.yaml; do
   cat "$crd" >> "$CRDS_FILE_PATH"
-  cat "$crd" >> "$HELM_CRDS_FILE_PATH"
```
Comment (leseb) [inline build/crds/build-crds.sh:100]: How does `$HELM_CRDS_FILE_PATH` get generated now?

File: build/crds/build-crds.sh [hunk 2]
```diff
@@ -67,18 +67,18 @@ EOF
...
-  sed -i '1s/^/{{- if .Values.crds.enabled }}\n/' "$HELM_CRDS_FILE_PATH"
-  
   echo "Generating helm resources.yaml"
   {
+    # add header
+    echo "{{- if .Values.crds.enabled }}"
+    echo "{{- if semverCompare \">=1.16.0\" .Capabilities.KubeVersion.GitVersion }}"
+
+    # Add helm annotations to all CRDS and skip the first 4 lines of crds.yaml
+    "$YQ_BIN_PATH" w -d'*' "$CRDS_FILE_PATH" "metadata.annotations[helm.sh/resource-policy]" keep | tail -n +5
```
Comment (leseb) [inline build/crds/build-crds.sh:77]: Why is `$HELM_CRDS_FILE_PATH` gone?

Comment (travisn) [reply build/crds/build-crds.sh:77]: It's cut off in the diff view, but see line 88 where everything in this block is piped to `>>"$HELM_CRDS_FILE_PATH"`.

File: build/crds/build-crds.sh (hunk 1 above)
Comment (travisn) [reply build/crds/build-crds.sh:100]: answered above

File: build/crds/build-crds.sh (hunk 2 above)
Comment (leseb) [reply build/crds/build-crds.sh:77]: Actually, the magic is in the removal of the `-i` flag, missed it in the first review.

**Review by leseb — APPROVED** [review]

File: build/crds/build-crds.sh (hunk 2 above)
Comment (travisn) [reply build/crds/build-crds.sh:77]: Right, and this command also takes `$CRDS_FILE_PATH` as the input

**Pull request merged**